src/fondant/pipeline/lightweight_component.py

```diff
@@ -110,6 +126,41 @@ def validate_abstract_methods_are_implemented(cls):
                     msg,
                 )
 
+        def modify_consumes_spec(apply_consumes, consumes_spec):
+            """Modify fields based on the consumes argument in the 'apply' method."""
+            if apply_consumes:
+                for k, v in apply_consumes.items():
+                    if isinstance(v, str):
+                        consumes_spec[k] = consumes_spec.pop(v)
+                    elif isinstance(v, pa.DataType):
+                        pass
+                    else:
+                        msg = (
+                            f"Invalid data type for field `{k}` in the `apply_consumes` "
+                            f"argument. Only string and pa.DataType are allowed."
+                        )
+                        raise ValueError(
+                            msg,
+                        )
+            return consumes_spec
+
+        def filter_consumes_spec(python_component_consumes, consumes_spec):
```

Comment:
Can we move this out of the decorator? It is not called when applying the decorator, so I would move it out of the decorator scope.